libs/set-algebra/src/Control/Iterate/BaseTypes.hs

```diff
@@ -248,46 +245,11 @@ instance Iter Sett where
   isnull (Sett x) = Set.null x
   lookup k (Sett m) = if Set.member k m then Just () else Nothing
 
--- ================ Basic BiMap ================================
-
-instance Ord v => Basic (BiMap v) where
-  addkv (k, v) (MkBiMap f b) comb = MkBiMap (Map.insertWith (mapflip comb) k v f) (insertBackwards oldv newv k b)
-    where
-      (oldv, newv) = case Map.lookup k f of Nothing -> (v, v); Just v2 -> (v2, comb v2 v)
-  removekey k (m@(MkBiMap m1 m2)) =
-    -- equality constraint (a ~ v) from (BiMap a k v) into scope.
-    case Map.lookup k m1 of
-      Just v -> MkBiMap (Map.delete k m1) (retract v k m2)
-      Nothing -> m
-  domain (MkBiMap left _right) = Map.keysSet left
-  range (MkBiMap _left right) = Map.keysSet right
-
-instance Ord v => Iter (BiMap v) where
-  nxt (MkBiMap left right) =
-    Collect
-      ( \ans f ->
-          case Map.minViewWithKey left of
-            Nothing -> ans
-            Just ((k, v), nextm) -> f (k, v, MkBiMap nextm right) ans
-      )
-  lub key (MkBiMap forward backward) =
-    Collect
-      ( \ans f ->
-          case Map.splitLookup key forward of -- NOTE in Log time, we skip over all those tuples in _left
-            (_left, Just v, right) -> f (key, v, MkBiMap right backward) ans
-            (_left, Nothing, right) | Map.null right -> ans
-            (_left, Nothing, right) -> f (k, v, MkBiMap m3 backward) ans
-              where
-                ((k, v), m3) = Map.deleteFindMin right
-      )
-  isnull (MkBiMap f _g) = isnull f
-  lookup x (MkBiMap left _right) = Map.lookup x left
-  haskey k (MkBiMap left _right) = haskey k left
-
 -- ============== Basic Map =========================
 
 instance Basic Map.Map where
-  addkv (k, v) m comb = Map.insertWith (mapflip comb) k v m
+  -- Data.Map uses(\ new old -> ...) while our convention is (\ old new -> ...)
+  addkv (k, v) m comb = Map.insertWith (flip comb) k v m
```

Comment:
@JaredCorduan  My only concern in this PR is this - so basically this `mapflip` defined in the BiMap was used also in the implementation of the map. Here is the implementation: https://github.com/input-output-hk/cardano-ledger/pull/3089/commits/e38e4048065c54a639593001b1ec66cc5ca6232b#diff-528417e48d8c20cf8e058ea8ec356378763095d96fc6841d76c65415772255c3L145

It seems to me that it's a simple `flip`, but would you mind double checking my logic?

Reply:
`mapflip` looks like a less general version of `flip` that doesn't have anything to do with maps :laughing: . LGTM!

Reply:
Thanks for confirming!